### DISCUSSION TITLE
home assistant custom icons

### DISCUSSION THREAD

**Discussion by akusokuzan** [StreamController/StreamController#278] [Q&A]
picked up a stream deck and thank you so much for this software it works amazingly well so many features i need a manual! Can someone please tell me how to do custom icons on home assistant based on state? I click custom icons i select the attribute the operator and the value but im completely lost on when is says then show icon. Ive tried the icon file location but I'm not sure how to select my icon. It has to be super simple and im just missing it. I'm not sure what to put in the then show box. I know im gonna feel stupid when someone tells me.

**Comment by Cfomodz** [comment]
In the "Then show icon" field, enter **Material Design Icon names only** - NOT file paths.

### Correct Format
- `lightbulb-on`
- `lightbulb-off`
- `door-open`
- `door-closed`
- `lock`
- `home`

### Incorrect Format
- ❌ `/path/to/icon.png`
- ❌ `mdi:lamp`
- ❌ `file:///home/user/icons/lamp.svg`

## Quick Example: Light Switch

**Customization 1** (Light ON):
- Attribute: `state`
- Operator: `==`
- Value: `on`
- Then show icon: `lightbulb-on`

**Customization 2** (Light OFF):
- Attribute: `state`
- Operator: `==`
- Value: `off`
- Then show icon: `lightbulb-off`

## Finding Icon Names

Browse icons at: **https://pictogrammers.com/library/mdi/**

Copy just the name (without "mdi:" prefix) and paste it in the "Then show icon" field.
